src/05-derivatives/03-imaging.md

```diff
@@ -5,14 +5,111 @@ extent and resolution.
+```
+
+The following metadata JSON fields are defined for preprocessed images:
+
+| **Key name**  | **Description**                                                                                                              |
+| ------------- | ---------------------------------------------------------------------------------------------------------------------------- |
+| SkullStripped | REQUIRED. Boolean. Whether the volume was skull stripped (non-brain voxels set to zero) or not.                              |
+| Resolution    | REQUIRED if `res` keyword present. Dictionary of Strings to Strings. Specifies the interpretation of the resolution keyword. |
+| Density       | REQUIRED if `den` keyword present. Dictionary of Strings to Strings. Specifies the interpretation of the density keyword.    |
+
+Example JSON file corresponding to
+`pipeline1/sub-001/func/sub-001_task-rest_run-1_space-MNI305_bold.json` above:
+
+```JSON
+{
+  "SkullStripped": true,
+  "Resolution": {
+    "hi": "Matched with high-resolution T1w (0.7mm, isotropic)",
+    "lo": "Matched with original BOLD resolution (2x2x3 mm^3)"
+  }
+}
+```
 
-| **Key name**  | **Description**                                                                                 |
-| ------------- | ----------------------------------------------------------------------------------------------- |
-| SkullStripped | REQUIRED. Boolean. Whether the volume was skull stripped (non-brain voxels set to zero) or not. |
+This would be equivalent to having two JSON metadata files, one
+corresponding to `res-lo`
+(`pipeline1/sub-001/func/sub-001_task-rest_run-1_space-MNI305_res-lo_bold.json`):
+
+```JSON
+{
+  "SkullStripped": true,
+  "Resolution": "Matched with original BOLD resolution (2x2x3 mm^3)"
+}
+```
+
+And one corresponding to `res-hi`
+(`pipeline1/sub-001/func/sub-001_task-rest_run-1_space-MNI305_res-hi_bold.json`):
+
+```JSON
+{
+  "SkullStripped": true,
+  "Resolution": "Matched with high-resolution T1w (0.7mm, isotropic)"
+}
+```
+
+Example of CIFTI files (i.e., a file that combines regularly sampled data
+and non-parametric surfaces) having both `res` and `den` keywords:
+
+```Text
+pipeline1/
+    sub-001/
+        func/
+            sub-001_task-rest_run-1_res-1_den-10k_bold.dtseries.nii
+            sub-001_task-rest_run-1_res-1_den-41k_bold.dtseries.nii
+            sub-001_task-rest_run-1_res-2_den-10k_bold.dtseries.nii
+            sub-001_task-rest_run-1_res-2_den-41k_bold.dtseries.nii
+            sub-001_task-rest_run-1_bold.dtseries.json
+```
+
+And the corresponding `sub-001_task-rest_run-1_bold.dtseries.json` file:
+
+```JSON
+{
+    "SkullStripped": true,
+    "Resolution": {
+        "1": "Matched with MNI152NLin6Asym 1.6mm isotropic",
+        "2": "Matched with MNI152NLin6Asym 2.0mm isotropic"
+    },
+    "Density": {
+        "10k": "10242 vertices per hemisphere (5th order icosahedron)",
+        "41k": "40962 vertices per hemisphere (6th order icosahedron)"
+    }
```

Comment:
in this example, how would spatial reference be included?

each density could have its own spatial reference.
each resolution could also have its own spatial reference.

Reply:
What do you mean?

This?
```Text
pipeline1/
    sub-001/
        func/
            sub-001_task-rest_run-1_den-10k_bold.dtseries.nii
            sub-001_task-rest_run-1_den-41k_bold.dtseries.nii
            sub-001_task-rest_run-1_bold.dtseries.json
```

Reply:
my question is how do i get the geometry (faces/triangles) of the `den-10k` vertices?

Reply:
That's a fair point, and seems to argue against this accumulated metadata mode.

I'm also unclear on what `SpatialReference` would point to for a grayordinate CIFTI. It could point to something like https://github.com/templateflow/tpl-fsLR/tree/f82f453, but that has more than three files in it.

We could allow `SpatialReference` to be an object, e.g.,

```
"SpatialReference": {
    "left": "/path/to/hemi-L.surf.gii",
    "right": "/path/to/hemi-R.surf.gii",
    "subcortical": "/path/to/subcortical_atlas.nii"
}
```

Reply:
I apologize it took me almost one week to mull over this.

- :+1: to @effigies' suggestion for the `SpatialReference`.
- Regarding @satra's question, the more I think it through, the more convinced I am that "greyordinates" are a new "manifold?" supporting a sampling scheme. Therefore, to be consistent they would allow only the density key. I believe HCP Pipelines made a good call for the valid values (e.g., den-170k, den-91k), based on the total number of samples. WDYT? This, in conjunction with the `atlas-fsLR` would allow for univocal designation of reference spaces without specifying a `SpatialReference`. Then, for Frankenstein grayordinates like _fMRIPrep_'s, one could always use @effigies's spec for `SpatialReference`.

Reply:
I think I agree with where things are going with spatial reference - although we may have the problem that the surface for the spatial reference can be multiple things..(the individuals pial surface, the individual white surface, the individual midthickness surface - as well as the versions of those within template flow). Although they _should_ all have the same density (# faces and # vertices) so maybe that is no big deal?

Reply:
@satra how would you go around the problem you pointed out?